packages/frontend-2/lib/user/composables/emails.ts

```diff
@@ -0,0 +1,179 @@
+  const { mutate: deleteMutation } = useMutation(settingsDeleteUserEmailMutation)
+  const { mutate: createMutation } = useMutation(settingsCreateUserEmailMutation)
+  const { mutate: verifyMutation } = useMutation(verifyEmailMutation)
+
+  const isEmailVerificationForced = useIsEmailVerificationForced()
+
+  // Simple array of all emails
+  const emails = computed(() => result.value?.activeUser?.emails ?? ([] as UserEmail[]))
+
+  // Helper computed properties for common queries
+  const unverifiedPrimaryEmail = computed(
+    () => emails.value.find((e) => e.primary && !e.verified) || null
+  )
+
+  const unverifiedEmails = computed(() => emails.value.filter((e) => !e.verified))
+
+  const addUserEmail = async (email: string) => {
+    const result = await createMutation({
+      input: { email }
+    }).catch(convertThrowIntoFetchResult)
+
+    if (result?.data) {
+      mixpanel.track('Email Added')
+      if (isEmailVerificationForced.value) {
+        navigateTo(verifyEmailRoute)
+      }
+      return true
+    }
+
+    const errorMessage = getFirstErrorMessage(result?.errors)
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Error adding email',
+      description: errorMessage
+    })
+    return false
+  }
+
+  const resendVerificationEmail = async (email: UserEmail) => {
+    const result = await resendMutation({
+      input: { id: email.id }
+    }).catch(convertThrowIntoFetchResult)
+
+    if (result?.data) {
+      triggerNotification({
+        type: ToastNotificationType.Success,
+        title: `Verification email sent to ${email.email}`
+      })
+      navigateTo(verifyEmailRoute)
+      return true
+    }
+
+    const errorMessage = getFirstErrorMessage(result?.errors)
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Error sending verification email',
+      description: errorMessage
+    })
+    return false
+  }
+
+  const deleteUserEmail = async (email: UserEmail, cancel = false) => {
+    const result = await deleteMutation({
+      input: { id: email.id }
+    }).catch(convertThrowIntoFetchResult)
+
+    if (result?.data) {
+      triggerNotification({
+        type: ToastNotificationType.Success,
+        title: `${cancel ? 'Cancelled adding email' : 'Deleted email'}`,
+        description: email.email
+      })
+
+      // If we're on the verify email page and there are no more unverified emails, redirect home
+      if (route.path === verifyEmailRoute && unverifiedEmails.value.length === 0) {
+        navigateTo(homeRoute)
+      }
+
+      return true
+    }
+
+    const errorMessage = getFirstErrorMessage(result?.errors)
+    triggerNotification({
+      type: ToastNotificationType.Danger,
+      title: 'Error deleting email',
+      description: errorMessage
+    })
+    return false
+  }
+
+  const verifyUserEmail = async (email: UserEmail, code: string) => {
```

Comment:
Maybe we can add some additional events here, so we can track the funnel of adding emails?
Email Added -> Email Verification Started -> Email Verified / Email Verified Cancelled, something like this maybe